meta-resin-common/recipes-core/chrony/files/chrony.conf

```diff
@@ -1,8 +1,11 @@
-initstepslew 30 0.resinio.pool.ntp.org
 server 0.resinio.pool.ntp.org iburst
 server 1.resinio.pool.ntp.org iburst
 server 2.resinio.pool.ntp.org iburst
 server 3.resinio.pool.ntp.org iburst
 driftfile /var/lib/chrony/drift
-makestep 1 3
+maxupdateskew 100
+makestep 1 -1
+logchange 1
+maxdistance 16
+hwtimestamp *
```

Comment:
* [`maxupdateskew`](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#maxupdateskew): the default is 100, and the docs say "Typical values for skew-in-ppm might be 100 for a dial-up connection to servers over a phone line, and 5 or 10 for a computer on a LAN." Was the `100` chosen because of that? Any idea how that might affect machines on LAN if at all?
* [makestep](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#makestep): as i understand the current setting is make a step, if the difference is more than 1s, and at any time? (I don't see the meaning of `-1` in the docs. Is that correct, really? Is it because the device is likely go offline/online? Or why allow steps any time? I know that many users hate steps/jumps, and this makes it more likely.
* [logchange](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#logchange): generate log if more than 1s change was done. That seems useful, though as I read the config, that is the default value as well? Without this directive, there are no logs at all, or are there logs with that default 1s threshold? Just to clarify.
* [`maxdistance`](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#maxdistance): the default value is 3 to reject sources. The larger value is due to the docs suggestion of "Setting `maxdistance` to a larger value can be useful to allow synchronisation with a server that only has a very infrequent connection to its sources and can accumulate a large dispersion between updates of its clock."? Or some other reason? It feels like the above shouldn't apply to us and should keep lower, as I read it the NTP *server* has infrequent connections to their clocks, not the client. Thus this change would allow using less precise servers. Is that the intention?
* [`hwtimestamp`](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#hwtimestamp): improved accuracy

Reply:
`maxupdateskew` The default is 1000. Various docs recommended 100 for dial-up/long latency and 5/10 for LAN connections. I chose 100 as we have long latency links such as 3g modems.

`makestep` syntax is `makestep threshold limit`. A negative value in the limit disables the limit. So step if time is out of sync greater than 1 second. And no limits on stepping. It used to only step 3 times. https://manpages.debian.org/testing/chrony/chrony.conf.5.en.html

`maxdistance` is a tricky one but mostly only aimed at windows ntp servers from what I could read in the faq. https://chrony.tuxfamily.org/faq.html#_using_a_windows_ntp_server

Reply:
@ZubairLK you didn't totally reply, I feel :)

* `maxupdateskew` you just repeat what I quoted. It's fine, but didn't address my question. (I guess if the default is even longer, then LAN devices shouldn't be affected too badly, though I wonder if longer is good for large latency, why not keep the default which is even longer)
* `makestep` repeat what I quoted, and didn't address the no limit aspect. Can that cause steps in time at any time during the device's online time? As that's not something that customers like, judging from support
* `maxdistance` sure, but didn't address whether it will make things worse, and is it really needed. There are 4 servers by default, chrony can just choose the lower distance ones in general, IMHO unlikely that all 4 will be high-distance-Windows-server.

I mean all fine by me, just I feel the chrony story is full of unintended side-effects and trying to make sure the config values we choose are based on solid ideas.